### ISSUE TITLE
Avoid overlapping private domains

### ISSUE THREAD

**Issue by jmcarp** [18F/cf-cdn-service-broker#67]
As @jcscottiii noticed today, users can't create overlapping private domains in separate cloud foundry organizations--for example, we can't create foo.bar.gov in one org and bob.foo.bar.gov in another. Hopefully this won't happen often, but it will probably happen more than never.

One possible workaround would be to have the broker automatically create all private domains in the same org, then share each private domain with the org where the service instance lives. This would be sort of magical, as @cnelson mentioned earlier, but it would save the user a step--and, more to the point, avoid the domain overlap issue. WDYT?

**Comment by jcscottiii** [comment]
@jmcarp I'm thinking about the case of abuse / user error. (which may be an extreme edge case) Let's say org A represents a.gov (they won't be able to own it though). Then someone in org B creates an app at `newsletter.a.gov`. While they org B won't be able to prove they own that via the CNAME changes, in CF, they would theoretically reserve that route/domain and then would require someone from org A to submit a ticket to us to verify and unmap the route from that rouge app in org B.

**Comment by jcscottiii** [comment]
Actually, the idea could work. Just a slight variation.

An additional plan that the domain owner org would use to regulate the other orgs for a specific to be shared to.

Create Service would be like:
```
cf create-service cdn-route shared-domain  18f-gov -c '{"domain": "18f.gov", "orgs":["gsa-18f-checklistomania", "gsa-18f-tock"]}'
```

and similarly, they could update the list which would check for differences and delete as such.

Also, we should disable the `private_domain_creation` feature flag. This way regular org managers can't create their private domains which could possibly overlap as well.

WDYT @jmcarp ?

**Comment by jmcarp** [comment]
Agreed that that's a problem, but it's a problem that already exists--as long as cloud foundry knows about external domains and orgs want to use potentially overlapping domains, it's going to be possible for users to block each other's routes.

**Comment by jmcarp** [comment]
Not sure that passing a list of orgs to share with is right, since the goal is to avoid the need for coordination between unrelated orgs and users.

**Comment by jcscottiii** [comment]
> between unrelated orgs and users.

yep, that word _unrelated_ is key. you're totally right.

**Comment by jmcarp** [comment]
For visibility, the list of solutions we've talked about include:

1. Auto-create all domains in a private org, then share with the user's org
    * If we can implement https://github.com/cloudfoundry/cloud_controller_ng/issues/780, we can prevent one user from squatting on another user's domain
1. Revert host forwarding, and use lambda to set `X-Forwarded-Host` to `Host`
1. Revert host forwarding, and allow users to set arbitrary custom headers
1. Document how to handle conflicts manually

I'm not sure how to prioritize this--I'm guessing we won't see non-18f domain conflicts in the wild anytime soon, but it's hard to know for sure. FWIW, I prefer the first option, which wouldn't take much time to implement.

cc @cnelson @mogul
